STIPS/STIP-002.md

```diff
@@ -0,0 +1,150 @@
+
+Now, when issuing/redeeming a SetToken which contain aToken as components, the above check fails. Cause of the failure:
+
+Aave accrues interest every block for an aToken. Internally it updates it's state only when the core functions of LendingPool are called, but view functions such as _aToken.balanceOf()_ always return the most up-to-date value, as they account for the current block number in calculating the balance. Basically they store a scaledBalance for every user and multiply it by the current liquidity index to determine the total aToken balance of the user.
+- aToken.balanceOf() returns scaledBalance * liquidityIndex
+- aToken.transfer(quantity) updates the scaledBalance by adding (quantity/liquidityIndex) to it.
+    - liquidityIndex = (timeDiff/secondsPerYear)*interestRate + 1, hence liquidityIndex is always > 1
+    - timeDiff = block.timestamp - lastUpdatedTimestamp
+    - lastUpdatedTimestamp = timestamp of last block in which there was an interaction with this aToken specific reserve, interactions include calling deposit, withdraw, repay, borrow, flashloan and swapBorrowRate functions on the reserve.
+
+Now in our ExplicitERC20#transferFrom function,
+- We call existingBalance = aToken.balanceOf(setToken), which sets 
+existingBalance = initialScaledBalance * index
+- By calling aToken.transfer(quantity), we update baseBalance to newScaledBalance
+newScaledBalance = initialScaledBalance + (quantity/index)
+- Then we set newBalance = aToken.balanceOf(setToken)
+    - newBalance = newScaledBalance * index
+- Finally we require, newBalance == existingBalance + quantity
+    - LHS = newBalance
+        - = newScaledBalance * index
+        - = (initialScaledBalance + (quantity/index)) * index
+        - = (initialScaledBalance * index) + (quantity/index)*index
+    - RHS = existingBalance + quantity
+        - = (initialScaledBalance * index) + quantity
+- Cancelling, initialScaledBalance * index from LHS and RHS,
+- We essentially require, (quantity/index) * index == quantity, with index > 1
+- Which will not always be true because solidity uint256 has limited precision and there is a chance of `(quantity/index) * index` to be greater/lesser than `quantity` by 1 wei.
+
+## Feasibility Analysis
+
+### Option 1:
+Instead of over-optimizing our checks to test for both undercollaterlization and overcollaterlization. We can just have checks to prevent undercollaterlization. This would prevent reverting in cases when `newBalance` of aToken is greater than `existingBalance + quantity` by 1 wei, and thus decrease the failure rate of issuance/redemption (maybe to 33%, cause we can consider less than, equal to, and greater than to be 3 equal cases). This option is the best considering time. The change is contained within the smart contracts and doesn't bubble up the stack leading to bigger changes in our backend.
```

Comment:
Would guess we'd decrease reverts by a much larger amount. Reverts will only happen if our `expected balance mod totalSupply == 0` and only then roughly 1/3 of the time (if we assume the rounding will lead to a difference from expectation of -1, 0, 1 an equal proportion of the time). But keep in mind the larger the `totalSupply` gets the the greater the range of return values for `expected balance mod totalSupply` and if we assume that each number in that range is equally likely (a uniform distribution) the probably of revert should tend towards 0 as supply increases